clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -18176,12 +18177,30 @@ Intrinsic::ID getDotProductIntrinsic(QualType QT, int elementCount) {
   return Intrinsic::dx_udot;
 }
 
+Intrinsic::ID getAllIntrinsic(const llvm::Triple::ArchType Arch) {
```

Comment:
this is a simple memchanism since we are only doing one intrinsic. 
Long term EmitHLSLBuiltinExpr may need to invoke a map using `llvm::Triple::ArchType Arch`  ie a arch map of  intrinsic maps. righting the same switch statement over and over again to pick the right intrinsics could become tedious.

```
spv_intrinsic = archMap[llvm::Triple::spirv][Builtin::BI__builtin_hlsl_elementwise_all] -->  Intrinsic::spv_all;
dx_intrinsic  = archMap[llvm::Triple::dxil][Builtin::BI__builtin_hlsl_elementwise_all] -->  Intrinsic::dx_all;
```

Reply:
We decided not to do this.